### PR TITLE
SIP370 Add guard to check for leach+vol -> negative min N

### LINKED ISSUE #370: Guard nitrogen limitation against negative available mineral N
**Title:** Guard nitrogen limitation against negative available mineral N

**Body:**

`checkNitrogenLimitation()` calculates:

```c
double availableMinN = envi.minN + nonUptakeDelta;
double reduction =
    (availableMinN / uptakeFrac + unclaimedStorage) / demand;
```

If combined volatilization and leaching exceed beginning mineral N plus mineralization during a timestep, `availableMinN` may become negative. This could produce a negative growth-reduction factor and therefore negative creation fluxes for reasons unrelated to negative NPP.

This appears to be a defensive corner case rather than an expected calibrated-model trajectory, so it need not block #359.

Suggested follow-up:

- Determine whether valid parameter bounds and timestep lengths already guarantee `availableMinN >= 0`.
- Otherwise, bound mineral-N losses or clip the supported uptake/growth calculation appropriately.
- Add a regression test with very low `minN` and combined volatilization/leaching exceeding available mineral N.
- Verify after a complete pool update that:
  - `minN >= 0`;
  - uptake, fixation, and the limitation factor are nonnegative;
  - material clamping is unnecessary;
  - whole-system N balance closes physically.

Relevant code: https://github.com/PecanProject/sipnet/blob/95f8f5e9091029253c9d8996b4c8f3ac78fd7fa2/src/sipnet/limitations.c#L76-L112

### PR DESCRIPTION
**Pull request by Alomir**
## Summary

- **What**: Guard against leaching + volatilization causing negative mineral N
- **Motivation**: This is an edge case, but still needs to be addressed

## Details

* Added guard in limitations.c
* Guard function called right before general N limitation check
* Unit test points added

## How was this change tested?
* Test points added for each case of excessive volatilization and leaching
* All other unit tests pass unmodified
* Smoke tests pass unmodified

## Related issues

- Fixes #370 

## Checklist

- [ ] Related issues are listed above. [PRs without an approved, related issue may not get reviewed](docs/CONTRIBUTING.md#propose-and-receive-feedback).
- [ ] PR title has the issue number in it ("[#<number>] \<concise description of proposed change>")
- [ ] Tests added/updated for new features (if applicable)
- [ ] Documentation updated (if applicable)
- [ ] `docs/CHANGELOG.md` updated with noteworthy changes
- [ ] Code formatted with `clang-format` (run `git clang-format` if needed)

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ sections to include in release notes:
 - Ability to create derived columns in `sipnet-view` (#355)
 - New `sipnet-debug-view` tool for visualizing SIPNET debug output files (#359)
 - Plant mortality check with event output on occurrence (#359)
+- Guard against negative mineral N due to volatilization and/or leaching (#375)
```
[the remark this answers is not in the document]
Comment (Alomir) [reply docs/CHANGELOG.md:38]: Sigh, we're back to this, are we?

File: tests/sipnet/test_modeling/testNitrogenCycle.c [hunk 2]
```diff
@@ -725,6 +725,94 @@ int testLeafTurnoverNResorption(void) {
...
+  // Leaching
+  // nLeaching = 1.0 * 0.8 * 0.5 = 0.4
+  params.nLeachingFrac = 0.5;
+  fluxes.drainage = 0.0;
+  double expNLeach = 0.0;
+
+  calcNVolatilizationFlux();
+  calcNLeachingFlux();
+  status |= checkVolAndLeachingFlux("volatilization", expNVol, expNLeach);
+  // Call the limit check should reduce nVol to 8 (enough to just exhaust nMin)
```
[the remark this answers is not in the document]
Comment (Alomir) [reply tests/sipnet/test_modeling/testNitrogenCycle.c]: True

File: src/sipnet/limitations.c [hunk 3]
```diff
@@ -113,10 +113,27 @@ static void checkNitrogenLimitation(void) {
...
+  double len = climate->length;
+  double pool = envi.minN + (fluxes.nMin + fluxes.eventMinN) * len;
+  double loss = (fluxes.nLeaching + fluxes.nVolatilization) * len;
+
+  if (loss > TINY && loss > pool) {
+    double reduction = pool / loss;
+    fluxes.nLeaching *= reduction;
+    fluxes.nVolatilization *= reduction;
+  }
+}
```
[the remark this answers is not in the document]
Comment (Alomir) [reply src/sipnet/limitations.c:129]: True

**Review by dlebauer — APPROVED** [review]
Looks great, fixes #370.
